torch/_lazy/__init__.py

```diff
@@ -20,3 +25,105 @@ def wait_device_ops(devices=None):
     if devices is None:
         devices = []
     torch._C._lazy._wait_device_ops(devices=devices)
+
+
+def create_lazy_process_group(store: dist.Store, rank: int, size: int, timeout: timedelta):
+    """
+    Create a ProcessGroup that is dedicated to handle lazy devices for torch.distributed.
+    Args:
+        store (dist.Store): key/value store accessible to all workers, used
+                            to exchange connection/address information.
+        rank (int): Rank of the current process (it should be a
+                    number between 0 and ``size``-1).
+        size (int): Number of processes participating in the job.
+        timeout (timedelta): Timeout for operations executed against
+                             the process group.
+    Returns:
+        A new LazyProcessGroup instance.
+    """
+    return LazyProcessGroup(store, rank, size, timeout)
+
+
+def lazy_comm_hook(state: object, bucket: dist.GradBucket) -> torch.futures.Future[torch.Tensor]:
+    """
+    A hook that compose LazyTensor with DistributedDataParallel. Ideally, one just need LazyProcessGroup.
+    Since dist._Work doesn't provide a way to set the value in its torch.futures.Future object, we have
+    to rely on this hook to set the right value.
+
+    Args:
+        state (object): Not used at all.
+        bucket: (dist.GradBucket): The bucket of gradients for all_reduce.
+
+    Returns:
+        A Future that contains a LazyTensor.
+    """
+    assert os.environ.get('LTC_TS_CUDA', False)
```

Comment:
whattt

Reply:
For CPU, the situation is a bit complicated. It has two related ProcessGroup, MPI and GLOO. Also I don't have any things that allow me to test CPU DDP. So, it's CUDA only now.